addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -493,6 +503,57 @@ def name_get(self):
                 for applicant in self
             ]
         return super().name_get()
+    @staticmethod
+    def _parse_ict_job_application(msg):
+        tree = html.fromstring(msg.get('body'))
+        text_elements = tree.xpath("//text()")
+        mail_index = 0
+
+        for text in text_elements:
+            if "@" in text:
+                break
+            mail_index += 1
+        try:
+            email = text_elements[mail_index]
+            name = text_elements[mail_index - 2]
+            first_name = text_elements[mail_index - 4]
+        except IndexError:
+            email = ''
+            name = ''
+            first_name = ''
+
+        return {
+                'name': msg.get('subject'),
+                'partner_name': f'{first_name} {name}',
+                'applicant_email_from': email,
+                'from_job_website': True
+            }
+
+    @staticmethod
+    def _parse_linkedin_application(msg):
+        tree = html.fromstring(msg.get('body'))
+        text_elements = tree.xpath("//text()")
+        href_elements = tree.xpath("//a/@href")
+
+        name_index = 0
```

Comment:
For this case where you only need the one element you could also do something like 
```
applicant_name = next((text_elements[index - 1] for index, text in enumerate(text_elements) if "·" in text), '')
```
though on second thought it might not be bad to stay consistent with the approach you used in the first parser.